pagerdutyplugin/resource_pagerduty_business_service.go

```diff
@@ -0,0 +1,230 @@
+}
+
+func (r *resourceBusinessService) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name":     schema.StringAttribute{Required: true},
+			"id":       schema.StringAttribute{Computed: true},
+			"html_url": schema.StringAttribute{Computed: true},
+			"self":     schema.StringAttribute{Computed: true},
+			"summary":  schema.StringAttribute{Computed: true},
+			"description": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString("Managed by Terraform"),
+			},
+			"type": schema.StringAttribute{
+				Optional:           true,
+				Computed:           true,
+				Default:            stringdefault.StaticString("business_service"),
+				DeprecationMessage: "This will change to a computed resource in the next major release.",
+				Validators:         []validator.String{stringvalidator.OneOf("business_service")},
+			},
+			"point_of_contact": schema.StringAttribute{Optional: true},
+			"team":             schema.StringAttribute{Optional: true},
+		},
+	}
+}
+
+func (r *resourceBusinessService) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan resourceBusinessServiceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	businessServicePlan := buildPagerdutyBusinessService(ctx, &plan)
+
+	err := helperResource.RetryContext(ctx, 5*time.Minute, func() *helperResource.RetryError {
+		log.Printf("[INFO] Creating PagerDuty business service %s", plan.Name)
+		bs, err := r.client.CreateBusinessServiceWithContext(ctx, businessServicePlan)
+		if err != nil {
+			return helperResource.NonRetryableError(err)
+		} else if bs != nil {
+			businessServicePlan.ID = bs.ID
+		}
+		return nil
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error calling CreateBusinessServiceWithContext",
+			err.Error(),
+		)
+		return
+	}
+
+	err = helperResource.RetryContext(ctx, 5*time.Minute, func() *helperResource.RetryError {
+		businessService, err := r.client.GetBusinessServiceWithContext(ctx, businessServicePlan.ID)
+		if err != nil {
+			return helperResource.RetryableError(err)
+		}
+		plan = buildResourceBusinessServiceModel(businessService)
+		return nil
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error calling GetBusinessServiceWithContext",
+			err.Error(),
+		)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *resourceBusinessService) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state resourceBusinessServiceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	log.Printf("[INFO] Reading PagerDuty business service %s", state.ID)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	err := helperResource.RetryContext(ctx, 5*time.Minute, func() *helperResource.RetryError {
```

Comment:
Could you please refactor this into a re-usable function to use it here and in `Create`?